provider/pkg/provider/container.go

```diff
@@ -0,0 +1,327 @@
+
+		for _, key := range updateOrder {
+			updateHandler, ok := nodepoolUpdateHandlers[key]
+			if !ok {
+				logging.V(9).Infof("[%s] No handler found for field: %q", label, key)
+				// TODO This should be an error once all the targetted mutations are in place!
+				continue
+			}
+			for k := range detailedDiff {
+				logging.V(9).Infof("[%s] processing diff on field: %q", label, k)
+				var diffPropPath, propPath resource.PropertyPath
+				diffPropPath, err = resource.ParsePropertyPath(k)
+				if err != nil {
+					err = fmt.Errorf("failed to parse property path: %q: %w", k, err)
+					break
+				}
+				propPath, err = resource.ParsePropertyPath(key)
+				if err != nil {
+					err = fmt.Errorf("failed to parse property path: %q: %w", k, err)
+					break
+				}
+				if propPath.Contains(diffPropPath) {
+					logging.V(9).Infof("[%s] processing diff for %q with handler for %q",
+						label, diffPropPath.String(), propPath.String())
+					logging.V(9).Infof("[%s] waiting for nodepool to reach resting state", label)
+					_, err = waitForNodepoolRestingState(providerInstance, res, inputs, oldState)
+					if err != nil {
+						break
+					}
+					logging.V(9).Infof("[%s] calling update handler", label)
+					err = updateHandler(providerInstance, urn, label, res, inputs, oldState)
+					if err != nil {
+						logging.V(9).Infof("[%s] failure updating: %+v", label, err)
+						break
+					}
+				}
+			}
+			if err != nil {
+				break
+			}
+		}
+	}
+	uri, readErr := res.Read.Endpoint.URI(inputs.Mappable(), oldState.Mappable())
+	if readErr != nil {
+		return nil, readErr
+	}
+	resp, err2 := readNodepoolStatus(providerInstance, uri)
+	if err2 != nil {
+		logging.V(9).Infof("[%s] Failed to read nodepool status: %v", label, err2)
+		return nil, fmt.Errorf("failed to retrieve nodepool status: %v. Previous error: %w", err2, err)
+	}
+	return resp, err
+}
+
+func readNodepoolStatus(
+	p *googleCloudProvider,
+	uri string,
+) (map[string]interface{}, error) {
+	// Read the current state of the resource from the API.
+	newState, err := retryRequest(p.client, "GET", uri, nil)
+	if err != nil {
+		return nil, fmt.Errorf("error fetching nodepool status: %w", err)
+	}
+	return newState, nil
+}
+
+func waitForNodepoolRestingState(p *googleCloudProvider,
+	res *resources.CloudAPIResource,
+	inputs,
+	oldState resource.PropertyMap,
+) (map[string]interface{}, error) {
+	ctx, cancelFunc := context.WithTimeout(context.Background(), 15*time.Minute)
+	defer cancelFunc()
+	retryPolicy := backoff.Backoff{
+		Min:    1 * time.Second,
+		Max:    15 * time.Second,
+		Factor: 1.5,
+		Jitter: true,
+	}
+
+	uri, err := res.Read.Endpoint.URI(inputs.Mappable(), oldState.Mappable())
+	if err != nil {
+		return nil, err
+	}
+
+	for {
+		resp, err := readNodepoolStatus(p, uri)
+		if err != nil {
+			return nil, err
+		}
+
+		status, hasStatus := resp["status"].(string)
+		if hasStatus && status == "RUNNING" || status == "RUNNING_WITH_ERROR" || status == "ERROR" {
```

Comment:
Probably worth using a switch statement with a default case that warns if an unexpected status message was found

Reply:
I don't know if I want to strongly assert status here. I am mostly reflecting existing logic from the terraform provider here. I did pull it out to a function with a switch case though.